### PR TITLE
Set 'strict mode' for Angular

### PR DESCRIPTION
**Pull request by egaluza**

### CODE REVIEW

File: packages/devextreme-schematics/src/add-layout/files/src/app/shared/components/side-navigation-menu/side-navigation-menu.component.ts [hunk 1]
```diff
@@ -71,7 +72,7 @@ export class SideNavigationMenuComponent implements AfterViewInit, OnDestroy {
   }
 
   ngAfterViewInit() {
-    events.on(this.elementRef.nativeElement, 'dxclick', (e) => {
+    events.on(this.elementRef.nativeElement, 'dxclick', (e: object) => {
```
Comment (dxvladislavvolkov) [inline packages/devextreme-schematics/src/add-layout/files/src/app/shared/components/side-navigation-menu/side-navigation-menu.component.ts:75]: Can we use Event type here?

File: packages/devextreme-schematics/src/add-layout/files/src/app/not-authorized-container.ts [hunk 2]
```diff
@@ -30,13 +30,15 @@ export class NotAuthorizedContainerComponent {
...
     }
+    return '';
   }
 
   get description() {
     const path = this.router.url.split('/')[1];
     switch (path) {
       case 'reset-password': return 'Please enter the email address that you used to register, and we will send you a link to reset your password via Email.';
     }
+    return '';
```
Comment (dxvladislavvolkov) [inline packages/devextreme-schematics/src/add-layout/files/src/app/not-authorized-container.ts:41]: Use the default of switch

File: packages/devextreme-schematics/src/add-layout/files/src/app/shared/components/change-password-form/change-password-form.component.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ import { AuthService } from '../../services';
 export class ChangePasswordFormComponent implements OnInit {
   loading = false;
   formData: any = {};
-  recoveryCode: string;
+  recoveryCode: string|null;
```
Comment (dxvladislavvolkov) [inline packages/devextreme-schematics/src/add-layout/files/src/app/shared/components/change-password-form/change-password-form.component.ts:18]: Add backspaces

Comment (egaluza) [reply packages/devextreme-schematics/src/add-layout/files/src/app/shared/components/change-password-form/change-password-form.component.ts:18]: Added

File: packages/devextreme-schematics/src/add-layout/files/src/app/not-authorized-container.ts (hunk 2 above)
Comment (egaluza) [reply packages/devextreme-schematics/src/add-layout/files/src/app/not-authorized-container.ts]: Rewrote with default

File: packages/devextreme-schematics/src/add-layout/files/src/app/shared/components/side-navigation-menu/side-navigation-menu.component.ts (hunk 1 above)
Comment (egaluza) [reply packages/devextreme-schematics/src/add-layout/files/src/app/shared/components/side-navigation-menu/side-navigation-menu.component.ts]: Yes, change by Event

**Review by LazyLahtak — APPROVED** [review]

**Pull request merged**